subsys/net/lib/dns/resolve.c

```diff
@@ -1356,6 +1356,9 @@ int dns_resolve_name(struct dns_resolve_context *ctx,
 	if (!ctx || !query || !cb) {
 		return -EINVAL;
 	}
+	if (type != DNS_QUERY_TYPE_A && type != DNS_QUERY_TYPE_AAAA) {
+		return -EINVAL;
+	}
```

Comment:
If the query type would be `DNS_RR_TYPE_ANY`, then we could return either address. This is probably something to think in the future and not really relevant in this PR though.